// nativescript-facebook/index.ios.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var applicationModule = require("tns-core-modules/application");
var login_button_1 = require("./ui/login-button");
exports.LoginButton = login_button_1.LoginButton;
var login_manager_1 = require("./login-manager");
exports._registerLoginCallback = login_manager_1._registerLoginCallback;
exports._registerLogoutCallback = login_manager_1._registerLogoutCallback;
exports.onLoginCallback = login_manager_1.onLoginCallback;
exports.onLogoutCallback = login_manager_1.onLogoutCallback;
exports.init = login_manager_1.init;
exports.requestPublishPermissions = login_manager_1.requestPublishPermissions;
exports.requestReadPermissions = login_manager_1.requestReadPermissions;
exports.login = login_manager_1.login;
exports.logout = login_manager_1.logout;
exports.getCurrentAccessToken = login_manager_1.getCurrentAccessToken;
var login_behavior_1 = require("./login-behavior");
exports.LoginBehavior = login_behavior_1.LoginBehavior;
var BaseDelegate = (function (_super) {
    __extends(BaseDelegate, _super);
    function BaseDelegate() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BaseDelegate.prototype.applicationDidFinishLaunchingWithOptions = function (application, launchOptions) {
        return FBSDKApplicationDelegate.sharedInstance().applicationDidFinishLaunchingWithOptions(application, launchOptions);
    };
    BaseDelegate.prototype.applicationOpenURLSourceApplicationAnnotation = function (application, url, sourceApplication, annotation) {
        return FBSDKApplicationDelegate.sharedInstance().applicationOpenURLSourceApplicationAnnotation(application, url, sourceApplication, annotation);
    };
    BaseDelegate.prototype.applicationDidBecomeActive = function (application) {
        FBSDKAppEvents.activateApp();
    };
    return BaseDelegate;
}(UIResponder));
BaseDelegate.ObjCProtocols = [UIApplicationDelegate];
function BaseDelegate_tsickle_Closure_declarations() {
    BaseDelegate.ObjCProtocols;
}
applicationModule.ios.delegate = BaseDelegate;
